# data set Titanic


## importamos las librerias necesarias para el data set 

In [109]:
import pandas as pd

## 1. analisis inicial 

### leemos el archivo csv train.csv


In [110]:
# data set cargado en crudo
df_raw= pd.read_csv('train.csv')

# data set copia para editarlo

df_procesado = df_raw.copy()

### mostramos las dimensiones del data set


In [111]:
df_procesado.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


### df.info()

despues de ejecutar el df.info() obtuvimos la siguiente información 
tenemos 12 columnas las cuales estan dividas en: id del pasajero, sobrevivido, clase en la que estaba (clase siendo referencia a la clase social en la que se estaba alojando el tripulante), que edad tenia, si viajaba con un acompañante o no, el numero de ticket, el puerto de embarque, tarifa, sexo, nombre, cuantos padres o hijos viajaban en la embarcacion.

### ¿cuantos datos nulos tenemos?

In [112]:
df_procesado.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

en las casillas donde se muestran datos vacios son las siguientes: 
1. Age con 177 datos nulos 
2. cabin con 687 datos nulos
3. embarked con 2 datos nulos

### df.head

con df.head() nos muestra los primeros 5 registros del data set 

In [113]:
df_procesado.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## 2. estrategia para la columna Age (edad)

se utilizara la mediana por que es la que permite encontrar el valor en el centro y para los datos que tenemos es la mas optima 

In [114]:
# Seleccionas la columna 'Age' y luego calculas su mediana
mediana_edad = df_procesado['Age'].median()

print(f"La mediana de la edad es: {mediana_edad} años ")

 # Es un atajo para df['Age'].median()
# mediana_edad = df.Age.median()

 #print(f"La mediana de la edad es: {mediana_edad}")

La mediana de la edad es: 28.0 años 


#### rellenar los valores faltantes



In [115]:
df_procesado['Age'].fillna(value=mediana_edad, inplace = True)

C:\Users\PC\AppData\Local\Temp\ipykernel_19292\1735000010.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_procesado['Age'].fillna(value=mediana_edad, inplace = True)


In [116]:
df_procesado['Age'].isnull().sum()

np.int64(0)

mostramos nuevamente los datos del dataSet despues de aplicar los cambios de llenar los valores faltantes en la columna Age con la mediana 

In [117]:
df_procesado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


## 3. estrategia de la columna cabin (camarote)

ya que la columna camarote puede tener relacion con la columna de la clase social, por lo cual no es muy recomendable borrarla 

lo que se va hacer es crear una nueva columna llamada tiene_camarote donde el valor 1 significa que si hay un dato en el camarote, donde 0 sea igual a que no tiene datos en el camarote si era nulo (null)

In [118]:
df_procesado['tiene_Camarote'] = df_procesado['Cabin'].notnull().astype(int)

In [119]:
df_procesado.info()

df_procesado.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   PassengerId     891 non-null    int64  
 1   Survived        891 non-null    int64  
 2   Pclass          891 non-null    int64  
 3   Name            891 non-null    object 
 4   Sex             891 non-null    object 
 5   Age             891 non-null    float64
 6   SibSp           891 non-null    int64  
 7   Parch           891 non-null    int64  
 8   Ticket          891 non-null    object 
 9   Fare            891 non-null    float64
 10  Cabin           204 non-null    object 
 11  Embarked        889 non-null    object 
 12  tiene_Camarote  891 non-null    int64  
dtypes: float64(2), int64(6), object(5)
memory usage: 90.6+ KB


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  tiene_Camarote  
0      0         A/5 21171   7.2500   NaN        S               0  
1      0          PC 17599  71.2833   C85        C               1  
2      0  STON/O2. 3101282   7.9250   NaN        S               0  
3      0            113803  53.1000  C123        S               1  
4      0            373450   8.0500   NaN        S               0

borramos la columna cabin para evitar ruido en la data

In [120]:
df_procesado.drop('Cabin',axis=1, inplace= True)

df_procesado.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Embarked  tiene_Camarote  
0      0         A/5 21171   7.2500        S               0  
1      0          PC 17599  71.2833        C               1  
2      0  STON/O2. 3101282   7.9250        S               0  
3      0            113803  53.1000        S               1  
4      0            373450   8.0500        S               0

## 4. estrategia para columna embarked

In [121]:
# saber cuales son los valores faltantes en la columna Embarked llenandola con la moda
moda_Embarked = df_procesado['Embarked'].mode()[0]

print(f" la moda es {moda_Embarked}")



 la moda es S


In [122]:
# llenar los datos faltantes con la moda 
df_procesado['Embarked'].fillna(value= moda_Embarked, inplace= True)


df_procesado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   PassengerId     891 non-null    int64  
 1   Survived        891 non-null    int64  
 2   Pclass          891 non-null    int64  
 3   Name            891 non-null    object 
 4   Sex             891 non-null    object 
 5   Age             891 non-null    float64
 6   SibSp           891 non-null    int64  
 7   Parch           891 non-null    int64  
 8   Ticket          891 non-null    object 
 9   Fare            891 non-null    float64
 10  Embarked        891 non-null    object 
 11  tiene_Camarote  891 non-null    int64  
dtypes: float64(2), int64(6), object(4)
memory usage: 83.7+ KB


C:\Users\PC\AppData\Local\Temp\ipykernel_19292\1549376155.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_procesado['Embarked'].fillna(value= moda_Embarked, inplace= True)


# fase 2 preparacion de datos para el analisis 